packages/block-library/src/columns/block.json

```diff
@@ -33,6 +33,9 @@
 			"padding": true,
 			"__experimentalDefaultControls": {
 				"padding": true
+			},
+			"__experimentalDefaultValues": {
```

Comment:
I wonder if it makes sense to use the proposed block.json changes from https://github.com/WordPress/gutenberg/pull/34180/files#diff-dba084b81a05295345ce9149af1a17ab073ddef43a393e851bcd3bdae85d3f32R81 for default values. (PR: **Prototype: merge block CSS with theme.json styles** #34180)

I don't know how far that particular investigation will go, but it could save some lines to "join forces"
cc @oandregal

Reply:
Oh, neat idea!

There's quite a lot of similarity here — for the case of `blockGap` we only want to use the value as a fallback value if none is set anywhere else, rather than always outputting it, so that might be a subtle difference between a default / fallback value and a value set in `blockStyles`. But keen for feedback or ideas on how to combine things if you have any @oandregal!

I suppose the caveat here is that this PR will likely need to be backported to 6.0 RC whereas it looks like #34180 is a bit more forward-looking and something potentially for 6.1?

Reply:
> I suppose the caveat here is that this PR will likely need to be backported to 6.0 RC whereas it looks like https://github.com/WordPress/gutenberg/pull/34180 is a bit more forward-looking and something potentially for 6.1?

Ah, very good point. 

`__experimentalDefaultValues` does also do what it says on the tin as you say.

Reply:
Why not just `blockGap: { default: "something" }` instead of `blockGap: true`? (We do these object configs for a lot of block supports already)

Reply:
The individual feature properties of the `spacing` support are currently `boolean | array` and the array is treated as a list of sides that are opted into, with logic like the following function (and the below `useIsDimensionsSupportValid`: https://github.com/WordPress/gutenberg/blob/2ce8e8d6ce5572518750c2e10cee9f2aa22d595c/packages/block-editor/src/hooks/dimensions.js#L167

It'd be a more complex change to refactor those props to also support an object (with a nested `sides` key), so I thought using a parallel object to set default values might make for a cleaner change given that this PR could be backported. There's a precedent in the Layout support, too, which has a default object and then within it, it specifies the properties:

https://github.com/WordPress/gutenberg/blob/56152c6037b46ff98de4e0247e5fb9eb91fca85d/packages/block-library/src/columns/block.json#L45

We could always rename this from `__experimentalDefaultValues` to `default`, and it'd look like the following:

```
		"spacing": {
			"blockGap": true,
			"margin": [ "top", "bottom" ],
			"padding": true,
			"__experimentalDefaultControls": {
				"padding": true
			},
			"default": {
				"blockGap": "2em"
			}
		},
```

What do you think?

Reply:
Also we'd have to find a model to define default side values too, right? Well, maybe not initially. 

> The individual feature properties of the spacing support are currently boolean | array and the array is treated as a list of sides that are opted into

It's pretty declarative, but I suppose we _could_ check for the existence of a `default` key (?) if we had to?

```js
	if (
		! support ||
		typeof support[ feature ] === 'boolean' ||
		!! support?.[ feature ]?.default
	) {
		return;
	}
```

Reply:
> There's a precedent in the Layout support, too, which has a default object and then within it, it specifies the properties:

Yeah, that's the one I was thinking about, since you can also do. `__experimentalLayout: true` (a defined object is the same thing as "true"). I don't think we should bother changing the others (padding, margin) now, but would be good to decide on an approach. I'd vote for the object approach personally but wouldn't mind the alternative used here.